movement/filesystem.c

```diff
@@ -166,6 +166,28 @@ bool filesystem_read_file(char *filename, char *buf, int32_t length) {
     return false;
 }
 
+bool filesystem_read_line(char *filename, char *buf, int32_t offset, int32_t length) {
```

Comment:
I understand the desire to return the number of bytes read, but not sure I understand "take offset as a point and do the update itself"? Take offset as a line offset rather than a byte offset?

Reply:
Sorry, point = pointer (those letters would have been useful. As in, pass the parameter:

```
int32_t *offset;
```

Then, we can be responsibility for updating the offset, and the caller just has to provide it on subsequent calls without modification.

cf strtok_r

Reply:
Uncertain about this, mainly because it means you have to read a file line by line every time. Let's say you have a file with lines of data (i.e. arrival times for each subway station on a line). Your watch face could stash the relevant station as an offset into the file, and know that it needs to read a line at offset X to get all the arrivals for Main Street. 

If I understand the suggestion correctly, the offset pointer would only be valid in the context of parsing lines sequentially, and then only once; whereas stating that you want to read a line at an arbitrary offset leverages the ability to seek to any set point and read bytes. This feels, to me, like a more generally useful form for the function to take.

Reply:
It would be the same value, just stored/used in the location you provided, so you could do:

```
int offset = 5;
filesystem_read_line(f, buf, &offset, length)
```

Then `*offset` would have the new offset.

i.e. the normal case is 'caller doesn't care what the offset is', but if the caller does care they still have access.

Reply:
I see, you’re suggesting to make offset an in/out parameter. That makes sense.

Reply:
Implemented this; *offset now gets set to the offset of the next line automatically, removing the need for the strlen call. Thanks for the suggestion!